metagraph/tests/annotation/test_annotated_dbg_helpers.cpp

```diff
@@ -2,65 +2,68 @@
 
 #include "../graph/all/test_dbg_helpers.hpp"
 
-#include "annotation/annotation_converters.hpp"
-#include "graph/annotated_graph_algorithm.hpp"
 #include "graph/representation/canonical_dbg.hpp"
+#include "graph/representation/hash/dbg_hash_fast.hpp"
+#include "graph/representation/bitmap/dbg_bitmap.hpp"
+
+#include "annotation/representation/column_compressed/annotate_column_compressed.hpp"
 
+// this next #include includes AnnotatedDBG. we need access to its protected
+// members to modify the underlying annotator
+#define protected public
+#include "annotation/annotation_converters.hpp"
 
 namespace mtg {
 namespace test {
 
 using namespace mtg::graph;
+using namespace mtg::annot;
 
 template <class Graph, class Annotation>
 std::unique_ptr<AnnotatedDBG> build_anno_graph(uint64_t k,
                                                const std::vector<std::string> &sequences,
                                                const std::vector<std::string> &labels,
-                                               DBGMode mode) {
+                                               DeBruijnGraph::Mode mode) {
     assert(sequences.size() == labels.size());
     auto graph = build_graph_batch<Graph>(k, sequences, mode);
     auto canonical = std::dynamic_pointer_cast<CanonicalDBG>(graph);
-    assert(!canonical || mode == DBGMode::CANONICAL_WRAPPER);
 
-    uint64_t max_index = canonical
-                            ? canonical->get_graph().max_index()
-                            : graph->max_index();
+    uint64_t max_index = canonical ? canonical->get_graph().max_index()
+                                   : graph->max_index();
 
     auto anno_graph = std::make_unique<AnnotatedDBG>(
         graph,
-        std::make_unique<annot::ColumnCompressed<>>(max_index)
+        std::make_unique<ColumnCompressed<>>(max_index)
```

Comment:
`AnnotatedDBG::check_compatibility` uses `max_index` from the underlying graph, so we need to set the size this way

Reply:
Then let's fix it there, so the wrapper returns the same max_index as its base graph.

Reply:
Having access to `max_index` is still needed for masked graphs (in case you use to mask a CanonicalDBG), so maybe it would be better to implement either `max_canonical_index`, or `get_base_graph` so one could call `get_base_graph().max_index()`

Reply:
I'd say that latter is more useful, since it provides a nice way to check the underlying type of a wrapped graph without having to guess how to unwrap it

Reply:
do we actually allow masking wrapped graphs? Doesn't it make more sense to mask the original primary graphs, and wrap the result when required?

Reply:
Even though we technically can mask a CanonicalDBG, in practice, the code checks for this case and unwraps the graph.

Reply:
I added `DeBruijnGraph::get_base_graph()`. This should simplify things quite a bit.